modules/text_server_adv/script_iterator.cpp

```diff
@@ -51,7 +51,7 @@ ScriptIterator::ScriptIterator(const String &p_string, int p_start, int p_length
 	}
 
 	int paren_size = PAREN_STACK_DEPTH;
-	ParenStackEntry *paren_stack = (ParenStackEntry *)memalloc(paren_size * sizeof(ParenStackEntry));
+	ParenStackEntry *paren_stack = static_cast<ParenStackEntry *>(memalloc(paren_size * sizeof(ParenStackEntry)));
```

Comment:
For the record, what are the kind of issues with using a C-style cast instead of `static_cast` or `dynamic_cast` in this context that cppcheck would report?

Reply:
`cppcheck` reports all C-casts, I guess the only purpose of these reports is to make intention of the cast more clear, so it's just a style change.